docs/releases.md

```diff
@@ -1,39 +1,160 @@
-- [ ] all tests pass (no exceptions)
-- [ ] run interop tests https://github.com/ipfs/interop#test-with-a-non-yet-released-version-of-go-ipfs
-- [ ] webui works (for most definitions of 'works') - Test the multiple pages and verify that no visible errors are shown.
+### Release Stage 1 - Internal testing
+
+When Release Stage 1, there is a features freeze for the release branch.
+
 - [ ] CHANGELOG.md has been updated
   - use `./bin/mkreleaselog` to generate a nice starter list
-- [ ] version string in `repo/version.go` has been updated
-- [ ] tag commit with vX.Y.Z
-- [ ] update release branch to point to release commit (`git merge vX.Y.Z`).
-- [ ] publish dist.ipfs.io
-- [ ] publish next version to https://github.com/ipfs/npm-go-ipfs
-
-## Post-Release
-- [ ] Bump version string in `repo/version.go` to `vX.Y.Z-dev`
-- [ ] Upload the final release to the github releases page: https://github.com/ipfs/go-ipfs/releases
-- Communication
-  - [ ] Create the release issue
-  - [ ] Announcements (both pre-release and post-release)
-    - [ ] Twitter
-    - [ ] IRC
-    - [ ] Reddit
-  - [ ] Blog post (at minimum, paste the changelog. optionally add context and thank contributors.)
-- [ ] Update HTTP-API Documentation on the Website using https://github.com/ipfs/http-api-docs
+- [ ] Automated Testing - Ensure that all tests are passing, this includes:
+  - [ ] unit
+  - [ ] sharness
+  - [ ] [interop](https://github.com/ipfs/interop#test-with-a-non-yet-released-version-of-go-ipfs)
+- [ ] Network Testing:
+  - [ ] test lab things
+- [ ] Infrastructure Testing:
+  - [ ] Deploy new version to a subset of Bootstrappers
+  - [ ] Deploy new version to a subset of Gateways
+  - [ ] Deploy new version to a subset of Preload nodes
+  - [ ] Collect metrics every day. Work with the Infrastructure team to learn of any hiccup
+- [ ] IPFS HTTP Client Libraries Testing:
+  - [ ] [JS](http://github.com/ipfs/js-ipfs-http-client)
+  - [ ] [Go](https://github.com/ipfs/go-ipfs-api)
+- [ ] IPFS Application Testing -  Run the tests of the following applications:
+  - [ ] WebUI
+  - [ ] IPFS Desktop
+  - [ ] IPFS Companion
+  - [ ] NPM on IPFS
+
+### Release Stage 2 - Invite _early testers_ to try it out
+
+- [ ] Reach out to the following IPFS _early testers_ for testing this release (check when no more problems have been reported):
+  - [ ] Infura
+  - [ ] Textila
+  - [ ] Pinata
+  - [ ] QRI
+- [ ] Run tests available in the following repos with the latest RC (check when all tests pass):
+  - [ ] [orbit-db](https://github.com/orbitdb/orbit-db)
+
+PSA: If you are a heavy user of `go-ipfs`, have developed a solid test infrastructure for your application and would love to help us would like to help us test `go-ipfs` release candidates, reach out to go-ipfs-wg@ipfs.io.
+
+### Release Stage 3 - Announce to the broader community
+
+- [ ] Documentation
+  - [ ] Ensure that CHANGELOG.md is up to date
+  - [ ] Ensure that README.md is up to date
```

Comment:
I past few years we had times when parts of `docs/config.md` and `docs/experimental-features.md` got out of date. Its the source of truth for many power users. It should be part of checklist:
```suggestion
  - [ ] Ensure that README.md is up to date
  - [ ] Ensure that `docs/config.md` and `docs/experimental-features.md` are up to date
```